src/useStorage.tsx

```diff
@@ -0,0 +1,153 @@
+import { epochAtSecondsFromNow, FALLBACK_EXPIRE_TIME } from './timeUtils'
+import { useBrowserStorage, useCookieStorage } from './Hooks'
+import { TInternalConfig } from './Types'
+
+interface IUseStorage {
+  refreshToken: string | undefined
+  setRefreshToken: (v: string | undefined) => void
+  refreshTokenExpire: number
+  setRefreshTokenExpire: (v: number) => void
+  token: string
+  setToken: (v: string) => void
+  tokenExpire: number
+  setTokenExpire: (v: number) => void
+  idToken: string | undefined
+  setIdToken: (v: string | undefined) => void
+  loginInProgress: boolean
+  setLoginInProgress: (v: boolean) => void
+  refreshInProgress: boolean
+  setRefreshInProgress: (v: boolean) => void
+}
+
+// useStorage is a hook which abstracts the storage mechanism for the auth tokens
+// crucially, when config.storage === 'cookie', it will use the useCookieStorage hook
+// instead of the useBrowserStorage hook to store the tokens in cookies instead of localStorage
+
+export const useStorage = (config: TInternalConfig): IUseStorage => {
```

Comment:
I would assume that `useStorage` was a slightly more generic version of `useBrowserStorage`, but it contains completely different logic. I think the naming should be changed, maybe something like `useAuthData` or something. I couldn't think of any good names, but maybe you can think of some options.